test/index.js

```diff
@@ -172,15 +172,15 @@ test('TypeClass', function() {
   //  Bar :: TypeClass
   var Bar = Z.TypeClass('my-package/Bar', [Foo], hasMethod('bar'));
 
-  eq(Foo['@@type'], 'sanctuary-type-classes/TypeClass');
+  eq(Foo.constructor['@@type'], 'sanctuary-type-classes/TypeClass');
```

Comment:
Indeed. My only concern is that `@@type` has very little to do with Fantasy Land, so a `@@type`-specific function would be a little out of place here. I'm considering a separate package, sanctuary-type, which would just provide this function. I like to idea of a readme dedicated to explaining the problem `@@type` solves and the simple algorithm used to determine a value's type identifier.